_studio/mfx_lib/genx/CMakeLists.txt

```diff
@@ -75,6 +75,60 @@ if(MFX_ENABLE_KERNELS)
     ${prefix}/genx_skl_simple_me_isa.cpp
   )
 
+  if (BUILD_KERNELS)
+    add_executable(embed_isa ${MSDK_GENX_ROOT}/mctf/src/embed_isa.c)
+    set_target_properties(embed_isa PROPERTIES
+      RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/mctf)
+
+    # Gets gen version from the platform name
+    # For example, for ${platform}=skl it will return ${genx}=gen9
+    function(get_genx platform genx)
+      if(${platform} STREQUAL "bdw")
+        set(genx "gen8" PARENT_SCOPE)
+      elseif(${platform} STREQUAL "skl")
+        set(genx "gen9" PARENT_SCOPE)
+      elseif(${platform} STREQUAL "icl")
+        set(genx "gen11" PARENT_SCOPE)
+      elseif(${platform} STREQUAL "icllp")
+        set(genx "gen11" PARENT_SCOPE)
+      endif()
+    endfunction()
+
+    # Generates CM kernel suitable for usage in C/C++ application sources
+    #   ${src} - input CM kernel source file
+    #   ${platform} - platform to generate kernel for
+    #   ${out_dir} - directory to generate output
+    # Generated outputs for ${src}=path/file.cpp, ${platform}=skl:
+    #   path/include/file_skl_isa.h
+    #   path/src/file_skl_isa.cpp
+    function(add_cmc_target src platform out_dir)
+      get_filename_component(tgt ${src} NAME_WE)
+      get_genx(genx ${platform})
+      add_custom_command(
+        OUTPUT ${tgt}_${platform}.isa
+        DEPENDS embed_isa ${src}
+        COMMAND ${CMC}
+          -march=${platform}
```

Comment:
Is better to use -Qxcm_jit_target=${platform} instead, because -march will not finalize the GenX target code as @artem-shaporenko mentioned.